### PR TITLE
Check for bad domain names

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/opts.go [hunk 1]
```diff
@@ -1522,6 +1522,9 @@ func parseJetStream(v interface{}, opts *Options, errors *[]error, warnings *[]e
 				opts.JetStreamMaxStore = mv.(int64)
 			case "domain":
 				opts.JetStreamDomain = mv.(string)
+				if subj := fmt.Sprintf(jsDomainAPI, opts.JetStreamDomain); !IsValidSubject(subj) {
```
Comment (kozlovic) [inline server/opts.go:1525]: In order to check for both file config and for users embedding the server, I recommend doing validation checks in server.go, `validateOptions()`

Comment (derekcollison) [reply server/opts.go:1525]: Good point..

**Review by matthiashanel — COMMENTED** [review]
agreeing with Ivan

File: server/jetstream_test.go [hunk 2]
```diff
@@ -11211,12 +11211,13 @@ func TestJetStreamOperatorAccounts(t *testing.T) {
 func TestJetStreamServerDomainBadConfig(t *testing.T) {
 	shouldFail := func(domain string) {
 		t.Helper()
-		conf := createConfFile(t, []byte(fmt.Sprintf("jetstream: {domain: \"%s\"}", domain)))
-		defer removeFile(t, conf)
-		if _, err := ProcessConfigFile(conf); err == nil {
-			t.Fatalf("Expected bad domain error, got %v", err)
+		opts := DefaultTestOptions
+		opts.JetStreamDomain = domain
+		if err := validateOptions(&opts); err == nil {
```
Comment (kozlovic) [inline server/jetstream_test.go:11216]: This is fine, but to make sure that you are getting the error you are checking for, I usually do:
```
if err := validateOptions(&opts); err == nil || !strings.Contains(err.Error(), "Invalid domain name") {
    t.Fatalf("Expected bad domain error, got %v", err)
}
```

**Review by kozlovic — APPROVED** [review]
LGTM, nit on checking for expected error..

**Review by kozlovic — CHANGES_REQUESTED** [review]
Sorry, I did not see that. Yes, when returning error, don't capitalize.

**Review by kozlovic — APPROVED** [review]

File: server/jetstream_test.go (hunk 2 above)
Comment (derekcollison) [reply server/jetstream_test.go]: Fixed.

**Comment by derekcollison** [comment]
ok should be good.

**Review by kozlovic — APPROVED** [review]
LGTM. Thanks for adding the error check.

**Pull request merged**